go-controller/pkg/ovn/zone_interconnect/interconnect.go

```diff
@@ -0,0 +1,524 @@
+	}
+
+	ts := &nbdb.LogicalSwitch{
+		Name: zic.networkTransitSwitchName,
+	}
+
+	_, err = libovsdbops.GetLogicalSwitch(zic.nbClient, ts)
+	if err != nil {
+		// Create transit switch
+		ts.OtherConfig = map[string]string{
+			"interconn-ts":             zic.networkTransitSwitchName,
+			"requested-tnl-key":        transitSwitchTunnelKey,
+			"mcast_snoop":              "true",
+			"mcast_flood_unregistered": "true",
+		}
+		if err := libovsdbops.CreateOrUpdateLogicalSwitch(zic.nbClient, ts); err != nil {
+			return fmt.Errorf("failed to create/update logical transit switch: %v", err)
+		}
+	}
+
+	var transitRouterPortMac net.HardwareAddr
+	var transitRouterPortNetworks []string
+	foundIPv4 := false
+	for _, ip := range nodeTransitSwitchPortIPs {
+		if utilnet.IsIPv4CIDR(ip) {
+			foundIPv4 = true
+			transitRouterPortMac = util.IPAddrToHWAddr(ip.IP)
+		} else {
+			if !foundIPv4 {
+				transitRouterPortMac = util.IPAddrToHWAddr(ip.IP)
+			}
+		}
+		transitRouterPortNetworks = append(transitRouterPortNetworks, ip.String())
+	}
+
+	remotePortAddr := transitRouterPortMac.String()
+	for _, tsNetwork := range transitRouterPortNetworks {
+		remotePortAddr = remotePortAddr + " " + tsNetwork
+	}
+
+	lspOptions := map[string]string{
+		"requested-tnl-key": strconv.Itoa(nodeID),
+	}
+	remotePortName := zic.getNetworkScopedName(types.TransitSwitchToRouterPrefix + node.Name)
+	if err := zic.addNodeLogicalSwitchPort(zic.networkTransitSwitchName, remotePortName, "remote", []string{remotePortAddr}, lspOptions); err != nil {
+		return err
+	}
+	// Set the port binding chassis.
+	if err := zic.setRemotePortBindingChassis(node.Name, remotePortName, chassisId); err != nil {
+		return err
+	}
+
+	if err := zic.addRemoteNodeStaticRoutes(node, nodeID, nodeTransitSwitchPortIPs); err != nil {
+		return err
+	}
+
+	// Cleanup the logical router port connecting to the transit switch for the remote node (if present)
+	// Cleanup would be required when a local zone node moves to a remote zone.
+	return zic.cleanupNodeClusterRouterPort(node)
+}
+
+func (zic *ZoneInterconnectHandler) addNodeLogicalSwitchPort(logicalSwitchName, portName, portType string, addresses []string, options map[string]string) error {
+	logicalSwitch := nbdb.LogicalSwitch{
+		Name: logicalSwitchName,
+	}
+
+	logicalSwitchPort := nbdb.LogicalSwitchPort{
+		Name:      portName,
+		Type:      portType,
+		Options:   options,
+		Addresses: addresses,
+	}
+	if err := libovsdbops.CreateOrUpdateLogicalSwitchPortsOnSwitch(zic.nbClient, &logicalSwitch, &logicalSwitchPort); err != nil {
+		return fmt.Errorf("failed to add logical port %s to switch %s, error: %v", portName, logicalSwitch.Name, err)
+	}
+	return nil
+}
+
+func (zic *ZoneInterconnectHandler) cleanupNodeClusterRouterPort(node *corev1.Node) error {
+	lrp := nbdb.LogicalRouterPort{
+		Name: zic.getNetworkScopedName(types.RouterToTransitSwitchPrefix + node.Name),
+	}
+	logicalRouterPort, err := libovsdbops.GetLogicalRouterPort(zic.nbClient, &lrp)
```

Comment:
ErrNotFound: false, for `DeleteLogicalRouterPorts` IMHO libovsdbops handles this?